### PR TITLE
editor-unshare-button

### LINKED ISSUE #2075: Button to unshare projects in the editor
**Describe the new addon**

Button to unshare projects in the editor because every single time we have to go to my stuff and unshare it.

### PR DESCRIPTION
**Pull request by FunctionalMetatable**
Resolves #2075

### Changes
adds an unshare button in the editor

### Reason for changes

unshare button
### Tests
tested locally

i should stop making too many PRs

### CODE REVIEW

File: addons/editor-unshare-button/userscript.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          type: "SET_COMMENT_FETCH_STATUS",
+          infoType: "project",
+          status: "FETCHING",
+        });
+
+        const project = redux.state.preview.projectInfo;
+        const author = project.author;
+        fetch(`/site-api/projects/all/${redux.state.preview.projectInfo.id}/`, {
+          headers: {
+            accept: "application/json, text/javascript, */*; q=0.01",
```
Comment (apple502j) [inline addons/editor-unshare-button/userscript.js:24]: Remove

**Comment by WorldLanguages** [comment]
Hmm, should this be a minor / "small feature" addon?

**Comment by TheColaber** [comment]
Hmm... It's pretty helpful, but can also be pretty dangerous. Accidently pressing the button can lead to you loosing a project's share date.

**Comment by FunctionalMetatable** [comment]
> Hmm... It's pretty helpful, but can also be pretty dangerous. Accidently pressing the button can lead to you loosing a project's share date.

Time to add confirm actions support

**Comment by WorldLanguages** [comment]
I think it should prompt you even if confirmations is disabled. Unsharing is very destructive.

**Comment by FunctionalMetatable** [comment]
> I think it should prompt you even if confirmations is disabled. Unsharing is very destructive.

Done

**Comment by RedGuy12** [comment]
setting true by default?



________________________________

**Comment by WorldLanguages** [comment]
Confirmations is off by default, so this addon (editor-unshare-button) should handle the confirmation by itself.

**Comment by apple502j** [comment]
Note that studio-tools leave button also has confirmations for leaving and is independent of the addon.

File: addons/editor-unshare-button/userscript.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+          button.classList.remove("sa-unshare-button");
+          redux.dispatch({
+            type: "UPDATE_PROJECT_INFO",
+            info: {
+              is_published: false,
+            },
+          });
+        });
+      },
+      { once: true }
```
Comment (WorldLanguages) [inline addons/editor-unshare-button/userscript.js:57]: Why once? PR can be merged if you explain @FunctionalMetatable or if you simply remove this.

Comment (FunctionalMetatable) [reply addons/editor-unshare-button/userscript.js]: removed it i forgot what it was for

**Pull request merged**